exec/java-exec/src/test/java/org/apache/drill/exec/expr/fn/impl/TestStringFunctions.java

```diff
@@ -114,6 +114,19 @@ public void testRegexpMatches() throws Exception {
   }
 
   @Test
+  public void testRegexpReplace() throws Exception {
+    testBuilder()
+        .sqlQuery("select regexp_replace(a, 'a|c', 'x') res1, regexp_replace(b, 'd', 'zzz') res2 " +
+                  "from (values('abc', 'bcd'), ('bcd', 'abc')) as t(a,b)")
+        .unOrdered()
+        .baselineColumns("res1", "res2")
+        .baselineValues("xbx", "bczzz")
+        .baselineValues("bxd", "abc")
+        .build()
+        .run();
+  }
+
```

Comment:
that's probably Calcite uses 'ISO-8859-1', which does not cover the special '€'.

Maybe you can use the following string? It has non-ascii character, and is covered by 'ISO-8859-1'.

select regexp_matches('München', 'München') res3 from (values(1));

[1] https://en.wikipedia.org/wiki/ISO/IEC_8859-1